R/merge_sce_list.R

```diff
@@ -0,0 +1,200 @@
+#'  - The resulting rowData slot rownames will be updated to `{rowname}-{sce_name}`.
+#'
+#'
+#' @param sce_list A list of SingleCellExperiment objects. The list may optionally
+#'  be named with batch information. If no names are provided, names will be generated
+#'  based on the SCE's index.
+#' @param batch_column A character value giving the resulting colData column name
+#'  to differentiate originating SingleCellExperiment objects. Often these values
+#'  are library IDs. Default value is `library_id`.
+#'  @param cell_id_column A character value giving the resulting colData colum name
+#'  to hold unique cell IDs formatted as their batch and original rowname. Default
+#'  value is `cell_id`.
+#' @param retain_coldata_cols A vector of colData columns which should be retained
+#'  in the the final merged SCE object.
+#' @param preserve_rowdata_cols A vector of column names that appear in originating
+#'  SCE objects' rowData slots which should not be renamed with
+#'  the given SCE object name or index is name is not given. These are generally
+#'  columns which are not specific to the given library's preparation or statistics.
+#'  For example, such a vector might contain items like "Gene", "ensembl_ids", etc.
+#'
+#' @return A SingleCellExperiment object containing all SingleCellExperiment objects
+#'   present in the inputted list
+#' @export
+#'
+#' @import SingleCellExperiment
+merge_sce_list <- function(sce_list = list(),
+                           batch_column = "library_id",
+                           cell_id_column = "cell_id",
+                           retain_coldata_cols = c("sum",
+                                                   "barcode",
+                                                   "detected",
+                                                   "total",
+                                                   "subsets_mito_sum",
+                                                   "subsets_mito_detected",
+                                                   "subsets_mito_percent",
+                                                   "miQC_pass",
+                                                   "prob_compromised"),
+                           preserve_rowdata_cols = NULL) {
+
+  # Check `sce_list`----------------------
+  if (is.null(names(sce_list))) {
+    warning(
+      glue::glue(
+        "Individual SCE objects in `sce_list` are not named, so batches will be
+        named based on their list index in the merged SCE object.")
+    )
+    names(sce_list) <- 1:length(sce_list)
+  }
+
+  if (length(sce_list) < 2) {
+    warning("There are fewer than two SCE objects in the provided `sce_list` so there is nothing to merge.")
+    # Early return:
+    return(sce_list)
+  }
+
+  # Check `retain_coldata_cols` ----------------
+  if (length(retain_coldata_cols) == 0) {
+    warning("All colData will be removed from the the merged SCE.
+     Please check that `retain_coldata_cols` was correctly specified.")
```

Comment:
```suggestion
    warning("All pre-existing colData will be removed from the the merged SCE.
     Please check that `retain_coldata_cols` was correctly specified.")
```
Just specifying that we are removing the existing `colData` since we do add the `cell_id` and `batch_column`.